analyze/analyses/dominators/emit.rs

```diff
@@ -0,0 +1,298 @@
+use std::collections::BTreeMap;
+use std::io;
+
+use csv;
+
+use formats::json;
+use formats::table::{Align, Table};
+use twiggy_ir as ir;
+use twiggy_opt as opt;
+use twiggy_traits as traits;
+
+use crate::analyses::{dominators::DominatorTree, utils};
+
+impl traits::Emit for DominatorTree {
+    #[cfg(feature = "emit_text")]
+    fn emit_text(&self, items: &ir::Items, dest: &mut io::Write) -> Result<(), traits::Error> {
+        let mut table = Table::with_header(vec![
+            (Align::Right, "Retained Bytes".to_string()),
+            (Align::Right, "Retained %".to_string()),
+            (Align::Left, "Dominator Tree".to_string()),
+        ]);
+
+        let opts = &self.opts;
+        let mut row = 0 as u32;
+
+        fn recursive_add_rows(
+            table: &mut Table,
+            items: &ir::Items,
+            dominator_tree: &BTreeMap<ir::Id, Vec<ir::Id>>,
+            depth: u32,
+            mut row: &mut u32,
+            opts: &opt::Dominators,
+            id: ir::Id,
+        ) {
+            assert_eq!(id == items.meta_root(), depth == 0);
+
+            if *row == opts.max_rows() {
+                return;
+            }
+
+            if depth > opts.max_depth() {
+                return;
+            }
+
+            if depth > 0 {
+                add_text_item(items, depth, id, table);
+            }
+
+            if let Some(children) = dominator_tree.get(&id) {
```

Comment:
If we got here, shouldn't we have discovered this `id` from something that would force it to be in the dominator tree? Perhaps we can just assert/unwrap here.

Reply:
hmmmm, I gave this one a try, and ran into this error:

```
thread 'main' panicked at '"\"function names\" subsection" item missing in dominator tree', src/libcore/option.rs:1008:5
```

This might be worth taking care of w/in a separate issue, but I'm totally with you that the `if let` here shouldn't be required

Reply:
I think maybe the dominator tree only includes an entry if the thing has children. I vaguely remember this... Anyways, carry on!

Reply:
That _sounds_ right to me too, looking at this:

https://github.com/rustwasm/twiggy/blob/master/ir/ir.rs#L212